components/translate/core/browser/resources/brave_translate.js

```diff
@@ -0,0 +1,118 @@
+// securityOrigin is predefined by translate_script.cc.
+const securityOriginHost = new URL(securityOrigin).host;
+
+// A method to rewrite URL in the scripts:
+// 1. change the domain to translate.brave.com;
+// 2. adjust static paths to use braveTranslateStaticPath.
+const rewriteUrl = (url) => {
+  try {
+    let new_url = new URL(url);
```

Comment:
should this check if the URL hostname is `translate.googleapis.com` or whatever domains are related to translate, then return original URL if not?

Reply:
@diracdeltas
1. All the  requests here related to translate, because we override XHR only in translate world. Otherwise the page will be obviously broken.

2. The scripts can make XHRs to different origin, includes gstatic and other. All of them should be redirected.